src/transport/SecureSessionMgr.cpp

```diff
@@ -335,6 +330,12 @@ void SecureSessionMgr::OnMessageReceived(const PacketHeader & packetHeader, cons
         state->SetPeerNodeId(packetHeader.GetSourceNodeId().Value());
     }
 
+    // Update peer address every time it changes
```

Comment:
This comment mostly restates the code. Consider rephrasing it to contain the description in this PR, something like:

```cpp
// TODO: once mDNS address resolution is available recosider if this is required
// This updates the peer address once a packet is received from a new address
// and serves as a way to auto-detect peer changing IPs.
```